shared-services/assessment_findings_write_policy.tf

```diff
@@ -0,0 +1,22 @@
+# ------------------------------------------------------------------------------
+# Create the IAM policy that allows write access to the assessment findings S3
+# bucket (specified by var.assessment_findings_bucket_arn).
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "assessment_findings_bucket_write" {
+  statement {
+    actions = [
+      "s3:PutObject",
+    ]
+    resources = [
+      "${var.assessment_findings_bucket_arn}/${var.assessment_findings_bucket_object_name_pattern}",
```

Comment:
Is there a dependency issue on fresh applies between this and https://github.com/cisagov/findings-data-import-terraform/pull/34? Does this bucket need to exist for this to apply correctly? You're using a bucket ARN here but what I'm assuming is the target bucket's configuration will now have a required variable that expects a role ARN which I'm assuming is for the `assessment_findings_bucket_write` role in this PR.

Reply:
You raise a good point, so I just did some testing.  I confirmed that Terraform will still let you apply this code and create this policy, even if the bucket ARN is for a bucket that does not exist.  It will however throw an error if you don't supply a string that looks like a bucket ARN.

Successfully applies:
`assessment_findings_bucket_arn = "arn:aws:s3:::this-is-not-a-real-bucket-that-exists-but-it-works-here"`
`assessment_findings_bucket_arn = "arn:aws:s3:::future-name-of-findings-bucket-that-hasnt-been-created-yet"`
`assessment_findings_bucket_arn = "arn:aws:s3:::*"`

Does not successfully apply:
`assessment_findings_bucket_arn = ""`
`assessment_findings_bucket_arn = "*"`

I think we are OK here as long as one decides in advance what one is going to name their findings bucket and applies this code before applying the code in `cisagov/findings-data-import-terraform` (which is what I'd expect when standing up a new COOL environment).

Reply:
I guess that leads me to ask should the variable instead be a bucket name instead of a fully formed ARN just to reduce complication in the variable?

Reply:
Good call - see https://github.com/cisagov/cool-accounts/pull/132/commits/eab3146f62fd8e0118ae55aba7b7450cd2b87b59.